### ISSUE TITLE
Merge slice with override no effect

### ISSUE THREAD

**Issue by wen-bing** [imdario/mergo#64]
```go
package test5

import (
	"testing"

	"github.com/imdario/mergo"
)

type Strudent struct {
	Name  string
	Books []string
}

func TestSliceNotOverride(t *testing.T) {
	var s1 = Strudent{
		Name:  "Jack",
		Books: []string{"a", "B"},
	}

	var s2 = Strudent{
		Name:  "Tom",
		Books: []string{"1"},
	}

	err := mergo.Merge(&s2, s1, mergo.WithOverride)
	if err != nil {
		t.Log("merge error")
		t.Fail()
	}
	if len(s2.Books) != 2 {
		t.Log("slice not ovrride")
		t.Fail()
	}
}
```
I expect s2's Books are overrided by s1's Books, however the test failed.

**Comment by ravernkoh** [comment]
I think the slice is appended.

**Comment by imdario** [comment]
@wen-bing, I checked your code and I confirm that the slice is being appended as @ravernkoh pointed. I think can be a bug but I need to check all the issues to see if it was reported before.

**Comment by imdario** [comment]
Ok, after reviewing other issues, Mergo works as intended. Merge will append to slices as it adds missing keys in maps. But you are right. WithOverride should work different, as it shouldn't append to destination slice.

**Comment by imdario** [comment]
It has been solved thanks to PR #70.

**Comment by imdario** [comment]
And it has been released as v0.3.3.

**Comment by wen-bing** [comment]
Awesome, Thanks!
